javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -6,48 +6,52 @@
 package javaslang.collection;
 
 import java.io.Serializable;
+import java.util.Objects;
 
 import static javaslang.collection.Arrays.*;
-import static javaslang.collection.NodeModifier.copyLeaf;
+import static javaslang.collection.NodeModifier.COPY_NODE;
 import static javaslang.collection.NodeModifier.updateLeafWith;
 
 /**
- * A `bit-mapped trie` is a very wide and shallow tree (for integer indices the depth will be ≤ 6).
+ * A `bit-mapped trie` is a very wide and shallow tree (for integer indices the depth will be `≤6`).
  * Each node has a maximum of `32` children (configurable).
  * Access to a given position is done by converting the index to a base 32 number and using each digit to descend down the tree.
  * Modifying the tree is done similarly, but along the way the path is copied, returning a new root every time.
- * Append inserts in the last leaf, or if the tree is full from the right, it adds another layer on top of it (the old root will be the first of the new one).
- * Prepend is done similarly, but an offset is needed, because adding a new top node (where the current root would be the last node of the new root)
+ * `Append` inserts in the last leaf, or if the tree is full from the right, it adds another layer on top of it (the old root will be the first of the new one).
+ * `Prepend` is done similarly, but an offset is needed, because adding a new top node (where the current root would be the last node of the new root)
  * shifts the indices by half of the current tree's full size. The `offset` shifts them back to the correct index.
+ * `Slice` is done by trimming the path from the root and discarding any `leading`/`trailing` values in effectively constant time (without memory leak, as in `Java`/`Clojure`).
  *
  * @author Pap Lőrinc
  * @since 2.1.0
  */
 final class BitMappedTrie<T> implements Serializable {
-    static int BRANCHING_BASE = 5;
-    static int branchingFactor()                   { return 1 << BRANCHING_BASE; }
+    static final int BRANCHING_BASE = 5;
+    static final int BRANCHING_FACTOR = 1 << BRANCHING_BASE;
+    static final int BRANCHING_MASK = -1 >>> -BRANCHING_BASE;
+
     static int firstDigit(int num, int depthShift) { return num >> depthShift; }
     static int digit(int num, int depthShift)      { return lastDigit(firstDigit(num, depthShift)); }
-    static int lastDigit(int num)                  { return num & (-1 >>> -BRANCHING_BASE); }
+    static int lastDigit(int num)                  { return num & BRANCHING_MASK; }
 
     private static final long serialVersionUID = 1L;
 
-    private static final BitMappedTrie<?> EMPTY = new BitMappedTrie<>(Arrays.empty(), 0, 0, 0);
+    private static final BitMappedTrie<?> EMPTY_BMT = new BitMappedTrie<>(Arrays.empty(), 0, 0, 0);
```

Comment:
Indeed, now I can rename it back. Previously it was used from `Vector`, which had its own `EMPTY`